hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -134,17 +134,17 @@ def __stoppable_test_function(self, data):
                 raise
 
     def test_function(self, data):
+        assert isinstance(data.observer, TreeRecordingObserver)
         self.call_count += 1
 
         try:
             self.__stoppable_test_function(data)
         except BaseException:
             self.save_buffer(data.buffer)
             raise
-        finally:
-            data.freeze()
-            self.note_details(data)
 
+        data.freeze()
+        self.note_details(data)
```

Comment:
What's the motivation for this change?

(I can't tell whether it's deliberate or accidental, so I mostly want to make sure it's not the latter.)

Reply:
Hmm. You know, neither can I. I recall there being some deliberate reason for it but I don't remember what and as a result I'm not sure if that reason still holds. I'll revert it and see what happens!

Reply:
It looks like the answer is that nothing bad happens. I suspect it was a change I made while debugging somethign else and that the difference is mostly harmless. I've reverted it.